source/base/data_out_base.cc

```diff
@@ -7741,19 +7748,49 @@ DataOutInterface<dim, spacedim>::write_vtu_in_parallel(
                                   vtk_flags,
                                   ss);
 
-    ierr = MPI_File_write_ordered(
-      fh, ss.str().c_str(), ss.str().size(), MPI_CHAR, MPI_STATUS_IGNORE);
+    // using prefix sum to find specific offset to write at.
+
+    const std::uint64_t size_on_proc = ss.str().size();
+    size_placeholder = size_on_proc;
+    std::uint64_t prefix_sum = 0;
+    ierr = MPI_Exscan(&size_on_proc, &prefix_sum, 1, MPI_CHAR, MPI_SUM, comm);
+    AssertThrowMPI(ierr);
+
+    ierr = MPI_Bcast(&header_size, 1, MPI_UNSIGNED, 0, comm);
+    AssertThrowMPI(ierr);
+
+    offset = static_cast<MPI_Offset>(header_size) * sizeof(char) + prefix_sum;
+
+    ierr = MPI_File_write_at_all(fh,
+                                 offset,
+                                 ss.str().c_str(),
+                                 ss.str().size(),
+                                 MPI_CHAR,
+                                 MPI_STATUS_IGNORE);
     AssertThrowMPI(ierr);
   }
 
   // write footer
-  if (myrank == 0)
+  if (myrank == rank - 1)
     {
       std::stringstream ss;
       DataOutBase::write_vtu_footer(ss);
       unsigned int footer_size = ss.str().size();
-      ierr = MPI_File_write_shared(
-        fh, ss.str().c_str(), footer_size, MPI_CHAR, MPI_STATUS_IGNORE);
+
+      ierr = MPI_Bcast(&offset, 1, MPI_UNSIGNED_LONG_LONG, rank - 1, comm);
+      AssertThrowMPI(ierr);
+
+      // std::uint64_t size_on_proc;
+      // ierr = MPI_Bcast(&size_on_proc, 1, MPI_INT64_T, rank-1, comm);
+      ierr = MPI_Bcast(&size_placeholder, 1, MPI_INT64_T, rank - 1, comm);
```

Comment:
What does this variable hold and why do you broadcast here? Again, all ranks need to participate in a broadcast, so this is wrong.